Micro16C.Web/src/Micro16C.Web.Client/Main.fs

```diff
@@ -0,0 +1,163 @@
+        |> PassManager.queueTransform Passes.deadCodeEliminationPass
+        |> PassManager.queueTransform Passes.instructionSimplifyPass
+        |> PassManager.queueTransform Passes.simplifyCFGPass
+        |> PassManager.queueTransform Passes.deadBitEliminationPass
+        |> PassManager.queueTransform Passes.jumpThreadingPass
+        |> PassManager.queueTransform Passes.constantPropagationPass
+        |> PassManager.queueTransform Passes.deadCodeEliminationPass
+        |> PassManager.queueTransform Passes.instructionSimplifyPass
+        |> PassManager.queueTransform Passes.simplifyCFGPass
+        |> PassManager.queueTransform Passes.deadBitEliminationPass
+        |> PassManager.queueTransform (debugModulePasses "After optimizations:")
+        |> PassManager.queueTransform Legalize.legalizeConstantsPass
+        |> PassManager.queueTransform Legalize.breakPhiCriticalEdgesPass
+        |> PassManager.queueTransform Passes.reorderBasicBlocksPass
+        |> PassManager.queueTransform (debugModulePasses "End of IR:")
+        |> PassManager.queueTransform GenAssembly.genAssemblyPass
+        |> PassManager.queueTransform GenAssembly.removeUnusedLabelsPass
+        |> PassManager.queueTransform GenAssembly.removeRedundantLabelsPass
+
+    Lex.tokenize text
+    |> Result.bind Parse.parse
+    |> Result.bind Sema.analyse
+    |> Result.map Codegen.codegen
+    |> Result.map (fun x -> passManager |> PassManager.run x)
+
+
+/// The Elmish application's model.
+type Model =
+    {
+        counter: int
+        error: string option
+        codeToCompile: string
+        generatedCode: string
+    }
+
+
+let initModel =
+    {
+        counter = 0
+        error = None
+        codeToCompile = "int r0 = R0;
+int r1 = R1;
+int mod;
+do
+{
+    mod = r1 % r0;
+    r1 = r0;
+    r0 = mod;
+}
+while(mod != 0);
+R2 = r1;"
+        generatedCode = ""
+    }
+
+
+/// The Elmish application's update messages.
+type Message =
+    | Increment
+    | Decrement
+    | SetCounter of int
+    | Error of exn
+    | ClearError
+    | CompileCode of string
+
+let update message model =
+    match message with
+
+    | Increment ->
+        { model with counter = model.counter + 1 }
+    | Decrement ->
+        { model with counter = model.counter - 1 }
+    | SetCounter value ->
+        { model with counter = value }
+
+    | Error exn ->
+        { model with error = Some exn.Message }
+    | ClearError ->
+        { model with error = None }
+    | CompileCode value ->
+        { model with codeToCompile = value; generatedCode = match compile value |> Result.map Assembly.asText with 
+                                                            | Ok value -> value
+                                                            | Result.Error s -> "Error compiling: " + s }
+
+type Main = Template<"wwwroot/main.html">
+
+let homePage model dispatch =
+    Main.Home()
+      .CodeToCompile(model.codeToCompile, fun v -> dispatch (CompileCode v))
```

Comment:
```suggestion
      .CodeToCompile(model.codeToCompile, CompileCode >> dispatch)
```

Reply:
I'm probably doing something wrong, but

![image](https://user-images.githubusercontent.com/10220080/141005427-4774f5f2-63b6-424e-a191-33fea2298802.png)

Reply:
Hmm that is odd. AFAIK this should compile and the gibberish of the error message above seems to indicate a compiler bug. Problem is that when compiling in Visual Studio this does not compile as well (it's not just IntelliSense). 

Oddly it compiles and runs without issues in JetBrains Rider. Why the IDE influences the compiler and build result??? No clue and it terrifies me. 

Lets use your original code then to stay compatible with Visual Studio.